src/sensor/parser-ping.cpp

```diff
@@ -23,12 +23,25 @@ void PingParserExt::parseBuffer(const QByteArray& data)
 Parser::ParserState PingParserExt::parseByte(const char byte)
 {
     PingParser::ParseState state = _parser.parseByte(byte);
-    if (state == PingParser::ParseState::NEW_MESSAGE) {
-        _rxMessage = _parser.rxMessage;
-        return Parser::ParserState::NEW_MESSAGE;
-    } else if (state == PingParser::ERROR) {
-        return Parser::ParserState::ERROR;
+    switch(state) {
+        case PingParser::ParseState::NEW_MESSAGE:
+            _rxMessage = _parser.rxMessage;
+            return Parser::ParserState::NEW_MESSAGE;
+        case PingParser::ParseState::ERROR:
+            return Parser::ParserState::ERROR;
+        case PingParser::ParseState::WAIT_START :
+        case PingParser::ParseState::WAIT_HEADER :
+        case PingParser::ParseState::WAIT_LENGTH_L:
+        case PingParser::ParseState::WAIT_LENGTH_H:
+        case PingParser::ParseState::WAIT_MSG_ID_L:
+        case PingParser::ParseState::WAIT_MSG_ID_H:
+        case PingParser::ParseState::WAIT_SRC_ID:
+        case PingParser::ParseState::WAIT_DST_ID:
+        case PingParser::ParseState::WAIT_PAYLOAD:
+        case PingParser::ParseState::WAIT_CHECKSUM_L:
+        case PingParser::ParseState::WAIT_CHECKSUM_H:
+            return Parser::ParserState::PARSING;
```

Comment:
I don't see the point here, the compiler already checks if `==` is a valid enum class and not a int.